scripts/generate-utility-data.ts

```diff
@@ -0,0 +1,253 @@
+  9734, // City of Jewett City; no electric
+
+  // VA
+  8198, // City of Harrisonburg; no electric
+]);
+
+/**
+ * Map from utility codes (numbers) or utility names (string) to replacement
+ * names. The keys must be exactly as they appear in the spreadsheet. The
+ * replacement names will be treated as if they were the name in the sheet.
+ */
+const OVERRIDES = new Map<string | number, string>([
+  // AZ
+  [1241, 'DixiePower'],
+  [15048, 'Electrical District No 2 Pinal County'],
+  [30518, 'Electrical District No 3 Pinal County'],
+  [15049, 'Electrical District No 4 Pinal County'],
+  [18280, 'Sulphur Springs Valley Electric Cooperative'],
+  [19728, 'UniSource Energy Services'],
+  [40165, 'DixiePower'],
+  [62683, "Tohono O'odham Utility Authority"],
+  [606953, 'UniSource Energy Services'],
+
+  // CO
+  [5997, 'Estes Park Power and Communications'],
+  [7300, 'Glenwood Springs Electric'],
+  [10066, 'K.C. Electric Association'],
+  [11256, 'Loveland Water and Power'],
+  [12860, 'Morgan County REA'],
+  [15257, 'Poudre Valley REA'],
+  [16603, 'San Luis Valley REC'],
+  [16616, 'San Isabel Electric'],
+
+  // CT
+  [7716, 'Groton Utilities'],
+  [13831, 'Norwich Public Utilities'],
+  [17569, 'South Norwalk Electric and Water'],
+
+  // NY
+  [1036, 'Con Edison'],
+  [1115, 'NYSEG'],
+  [1117, 'National Grid'], // Niagara Mohawk
+  [3249, 'Central Hudson Gas & Electric'],
+  [4226, 'Con Edison'],
+  [11811, 'Massena Electric'],
+  [13511, 'NYSEG'],
+  [14154, 'Orange & Rockland'],
+  [16183, 'Rochester Gas & Electric'],
+  [16549, 'Salamanca Board of Public Utilities'],
+
+  // RI
+  [1857, 'Block Island Power Company'],
+
+  // VA
+  [1186, 'Dominion Energy'],
+  [2248, 'BVU Authority'],
+  [4794, 'Danville Utilities'],
+  [6715, 'Franklin Municipal Power and Light'],
+  [13640, 'NOVEC'],
+  [15619, 'Radford Electric Department'],
+  [19876, 'Dominion Energy'],
+  [60762, 'BVU Authority'],
+  ['Virginia Electric & Power Co', 'Dominion Energy'],
+
+  // VT
+  [1061, 'Green Mountain Power'],
+  [7601, 'Green Mountain Power'],
+  [19791, 'Vermont Electric Coop'],
+  [27316, 'Stowe Electric Department'],
+]);
+
+/**
+ * A best-effort attempt to convert the name from the spreadsheet into a
+ * utility ID and a user-facing name, as we've done manually.
+ */
+function convertName(
+  name: string,
+  state: string,
+): { id: string; name: string } {
+  let cleaned = name;
+
+  // Remove the " - (XX)" suffix (state codes)
+  cleaned = cleaned.replace(/-? *\([A-Z]{2}\)$/, '').trim();
+  // Remove square-bracketed parts, which are parent companies not used as the
+  // customer-facing brand
+  cleaned = cleaned.replace(/\[.*\]/, '').trim();
+  // Remove "Inc" or "Co" suffix
+  cleaned = cleaned.replace(/,? *(I(nc)?|Co)\.?$/, '').trim();
+  // Remove "(for __ reporting)" parenthetical
+  cleaned = cleaned.replace(/\(.*reporting.*\)/i, '').trim();
+  // Spell out "Pwr", "Assn", "Elec", and "Coop"
+  // The dataset is inconsistent about this
```

Comment:
Without knowing how consistent the dataset is - I notice that some of these replacements require capitalization to match. Should we continue to assume all entries will have capitalization of names? (e.g. below, Pwr will only be cleaned to "Power" if matching exactly).

Reply:
As far as I can tell, the dataset is perfectly consistent about these abbreviations being capitalized. I can make these case-insensitive, but the replacement would still be titlecase. On the whole I think that's better than leaving an abbreviation in, because a human reviewer is more likely to notice something amiss?